lichess.js

```diff
@@ -0,0 +1,420 @@
+            json['best_players_loose_dates'] = best_players_loose_dates
+            return json;
+        }
+        catch(error) {
+            console.error(`Could not get worst and best games: ${error}`);
+        }
+    }
+
+    else {
+        return null
+    }
+}
+
+
+
+async function lichessAPIRequestStatistics(user){
+
+    if(user != 'N/A'){
+        var myHeaders = new Headers();
+        myHeaders.append('Content-Type', 'application/json');
+        
+        var myInit = {
+            method: 'GET',
+        };
+  
+        const new_request = new Request('https://lichess.org/api/user/'+user, myInit);
+        
+        try {
+            const new_response = await fetch(new_request);
+            if (!new_response.ok) {
+            throw new Error(`HTTP error: ${new_response.status}`);
+            }
+            const response = await new_response.json();
+            
+
+            player_rate = response['perfs']['bullet']['rating'];
+
+            progress_games = 12;
+
+            progress = response['perfs']['bullet']['prog'];
+
+
+
+            deviation = response['perfs']['bullet']['rd'];
+
+            games_number_all = response['count']['all'];
+
+            games_number_rate = response['count']['rated'];
+
+
+
+            let play_days  = Math.floor(response['playTime']['total'] / (3600 * 24));
+            let play_hours = Math.floor((response['playTime']['total'] - (play_days * 3600 * 24)) / 3600);
+            let play_minutes = Math.floor((response['playTime']['total'] - (play_days * 3600 * 24)  - (play_hours * 3600)) / 60);
+
+            play_time = play_days  + ' дней ' + play_hours + ' часов ' + play_minutes + ' минут ';
+
+
+
+            wins_number = "|" + response['count']['win'] + "|";
+
+            wins_number_percent = Math.floor(response['count']['win']/response['count']['all'] * 100) + "%";
+
+
+            loss_number = "|" + response['count']['loss'] + "|";
+
+
+            loss_number_percent = Math.floor(response['count']['loss']/response['count']['all'] * 100) + "%";
+
+            draws_number = "|" + response['count']['draw'] + "|";;
+
+            draws_number_percent = Math.floor(response['count']['draw']/response['count']['all'] * 100) + "%";
+
+
+
+            json = {};
+            json['player_rate'] = player_rate;
+            json['progress_games'] = progress_games
+            json['progress'] = progress
+            json['deviation'] = deviation
+            json['games_number_all'] = games_number_all
+            json['games_number_rate'] = games_number_rate
+            json['play_time'] = play_time
+            json['wins_number'] = wins_number
+            json['wins_number_percent'] = wins_number_percent
+            json['loss_number'] = loss_number
+            json['loss_number_percent'] = loss_number_percent
+            json['draws_number'] = draws_number
+            json['draws_number_percent'] = draws_number_percent
+            return json;

```

Comment:
u can constantly create an object with `{}` syntax.

Reply:
it's better to return json respons in ur api_service, than pass it to the DTO and parse it there.